File: assingnment_0.py
import os

def main():
    print(f"BEFORE FORKS ----------- PID = {os.getpid()} ---------- PPID = {os.getppid()} \n\n")

    os.fork()
    print(f"AFTER 1 FORK ----------- PID = {os.getpid()} ---------- PPID = {os.getppid()}")

    os.fork()
    print(f"AFTER 2 FORK ----------- PID = {os.getpid()} ---------- PPID = {os.getppid()}")

    os.fork()
    print(f"AFTER 3 FORK ----------- PID = {os.getpid()} ---------- PPID = {os.getppid()}")


if __name__ == "__main__":
    main()

File: concurrency/c/assingnment_0.c
#include <stdio.h>
#include <unistd.h>


int main() {
    printf("BEFORE FORKS ----------- PID = %d ---------- PPID = %d\n\n", getpid(), getppid());

    fork();
    printf("AFTER 1 FORK ----------- PID = %d ---------- PPID = %d\n", getpid(), getppid());

    fork();
    printf("AFTER 2 FORK ----------- PID = %d ---------- PPID = %d\n", getpid(), getppid());

    fork();
    printf("AFTER 3 FORK ----------- PID = %d ---------- PPID = %d\n", getpid(), getppid());

    return 0;
}